### PR TITLE
SCP-3030 Plugin versioning cli updated

### PR DESCRIPTION
**Pull request by NicoleYWDu**
Release Notes
-------------
Add `confluent connect custom-plugin version` commands for managing custom plugin versions

New Features
- support `confluent connect custom-plugin version create` 
- support `confluent connect custom-plugin version update` 
- support `confluent connect custom-plugin version list` 
- support `confluent connect custom-plugin version delete` 
- support `confluent connect custom-plugin version describe` 

Checklist
---------
- [ ] Leave this box unchecked if features are not yet available in production

What
----

References
----------

https://confluentinc.atlassian.net/browse/SCP-2167

Test & Review
-------------

### CODE REVIEW

File: internal/connect/command_custom_plugin_version.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	Plugin              string   `human:"Plugin" serialized:"plugin"`
+	Version             string   `human:"Version" serialized:"version"`
+	Name                string   `human:"Name" serialized:"name"`
+	Description         string   `human:"Description" serialized:"description"`
+	Cloud               string   `human:"Cloud" serialized:"cloud"`
+	ConnectorClass      string   `human:"Connector Class" serialized:"connector_class"`
+	ConnectorType       string   `human:"Connector Type" serialized:"connector_type"`
+	SensitiveProperties []string `human:"Sensitive Properties" serialized:"sensitive_properties"`
+	VersionNumber       string   `human:"Version Number" serialized:"version_number"`
+	IsBeta              string   `human:"Beta" serialized:"is_beta"`
```
Comment (brianstrauch) [inline internal/connect/command_custom_plugin_version.go:21]: Should this be a `bool`? The only options are "true" and "false", no? This matters since the YAML and JSON output will be different.

File: internal/connect/command_custom_plugin_version_create.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+func (c *customPluginCommand) newVersionCreateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a custom connector plugin version.",
+		Args:  cobra.NoArgs,
+		RunE:  c.createCustomPluginVersion,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: `Create custom connector plugin version for plugin "ccp-123456".`,
+				Code: "confluent connect custom-plugin version create --plugin ccp-123456 --plugin-file datagen.zip --version-number 0.0.1 --beta=false --sensitive-properties=property1,property2",
```
Comment (brianstrauch) [inline internal/connect/command_custom_plugin_version_create.go:36]: `--beta=false` is the default, right? I don't think you would ever do this.
```suggestion
				Code: "confluent connect custom-plugin version create --plugin ccp-123456 --plugin-file datagen.zip --version-number 0.0.1 --beta=true --sensitive-properties=property1,property2",
```

File: internal/connect/command_custom_plugin_version_list.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		Short: "List custom connector plugin versions for plugin.",
+		Args:  cobra.NoArgs,
+		RunE:  c.listVersions,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "List custom connector plugin versions for plugin",
+				Code: "confluent connect custom-plugin version list --plugin ccp-123456",
+			},
+		),
+	}
```
Comment (brianstrauch) [inline internal/connect/command_custom_plugin_version_list.go:23]:
```suggestion
	}
	
```

File: internal/connect/command_custom_plugin_version_update.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+)
+
+func (c *customPluginCommand) newVersionUpdateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update",
+		Short: "Update a custom connector plugin version metadata.",
+		Args:  cobra.NoArgs,
+		RunE:  c.updateVersion,
+		Example: examples.BuildExampleString(
+			examples.Example{
```
Comment (brianstrauch) [inline internal/connect/command_custom_plugin_version_update.go:22]: Can we add another example for updating the release notes? Let's make sure we add quotes since the release notes will probably contain spaces.

File: pkg/deletion/deletion.go [hunk 5]
```diff
@@ -47,6 +47,26 @@ func ValidateAndConfirmDeletion(cmd *cobra.Command, args []string, checkExistenc
 	return nil
 }
 
+func ValidateAndConfirmDeletionCustomPluginVersion(cmd *cobra.Command, args []string, checkExistence func(string) bool, resourceType, pluginId, versionId string) error {
```
Comment (brianstrauch) [inline pkg/deletion/deletion.go:50]: Why do we need a specialized function here?

File: test/fixtures/output/connect/custom-plugin/version/create-invalid-is-beta.golden [hunk 6]
```diff
@@ -0,0 +1 @@
+Error: only is-beta values "true" or "false" are allowed
```
Comment (brianstrauch) [inline test/fixtures/output/connect/custom-plugin/version/create-invalid-is-beta.golden:1]: Is this file still needed? I don't think this error is printed anymore.

File: test/fixtures/output/connect/custom-plugin/version/create-help.golden [hunk 7]
```diff
@@ -0,0 +1,24 @@
+Create a custom connector plugin version.
+
+Usage:
+  confluent connect custom-plugin version create [flags]
+
+Examples:
+Create custom connector plugin version for plugin "ccp-123456".
+
+  $ confluent connect custom-plugin version create --plugin ccp-123456 --plugin-file datagen.zip --version-number 0.0.1 --beta=false --sensitive-properties=property1,property2
```
Comment (brianstrauch) [inline test/fixtures/output/connect/custom-plugin/version/create-help.golden:9]: Are "property1,property2" actual property keys that we would see in production?

Comment (brianstrauch) [reply test/fixtures/output/connect/custom-plugin/version/create-help.golden:9]: Maybe `aws.key,aws.value` would be better here?

**Comment by NicoleYWDu** [comment]
> [CLA assistant check](https://cla-assistant.io/confluentinc/cli?pullRequest=2887) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/confluentinc/cli?pullRequest=2887) before we can accept your contribution.**1** out of **2** committers have signed the CLA.✅ NicoleYWDu❌ ajhingranYou have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/confluentinc/cli?pullRequest=2887) it.
@ajhingran would you mind to signed the CLA for the pr?

File: test/fixtures/output/connect/custom-plugin/version/create-invalid-is-beta.golden (hunk 6 above)
Comment (NicoleYWDu) [reply test/fixtures/output/connect/custom-plugin/version/create-invalid-is-beta.golden:1]: you are right, remove this file.

File: pkg/deletion/deletion.go (hunk 5 above)
Comment (NicoleYWDu) [reply pkg/deletion/deletion.go:50]: because we want user get the info for both  pluginId and versionId while they want to delete the plugin version.
User can have many plugins and each plugin can have many versions.

Comment (brianstrauch) [reply pkg/deletion/deletion.go:50]: Hey @sgagniere, as of v4 aren't we just prompting the user with "y/n" for every deleted resource?

**Comment by brianstrauch** [comment]
> @ajhingran would you mind to signed the CLA for the pr?

Alternatively you can mark yourself as a public member of the "confluentinc" GitHub organization.

File: test/fixtures/output/connect/custom-plugin/version/create-help.golden (hunk 7 above)
Comment (NicoleYWDu) [reply test/fixtures/output/connect/custom-plugin/version/create-help.golden:9]: it could be keys, tokens. let me modify as what we have for api doc:
```
"sensitive_config_properties": [
"passwords",
"keys",
"tokens"
],
```

**Comment by ajhingran** [comment]
Signed the CLA.

File: pkg/deletion/deletion.go (hunk 5 above)
Comment (sgagniere) [reply pkg/deletion/deletion.go:50]: Oops, missed this.

Yeah, all of our delete confirmation prompts are yes/no now.

Comment (NicoleYWDu) [reply pkg/deletion/deletion.go:50]: Hi @sgagniere , thanks for your reply! I created and merged a new [PR](https://github.com/confluentinc/cli/pull/2889) with y/n for deletion cmd.

Comment (sgagniere) [reply pkg/deletion/deletion.go:50]: I think this branch was started before v4, so there will need to be a merge.
